[CI] FieldExtractorIT BWC test failing with NPE

Looks like an NPE:
```
 org.elasticsearch.client.ResponseException: method [POST], host [http://[::1]:42189], URI [/_query?format=cbor&pretty=true&error_trace=true], status line [HTTP/1.1 500 Internal Server Error]	
    Warnings: [No limit defined, adding default limit of [500]]	
    ¿eerror¿jroot_cause¿dtypevnull_pointer_exceptionfreasonxCannot invoke "org.elasticsearch.xpack.esql.planner.Layout$ChannelSet.nameIds()" because the return value of "java.util.List.get(int)" is nullkstack_traceyorg.elasticsearch.ElasticsearchException$1: Cannot invoke "org.elasticsearch.xpack.esql.planner.Layout$ChannelSet.nameIds()" because the return value of "java.util.List.get(int)" is null	
    	at org.elasticsearch.server@8.12.2/org.elasticsearch.ElasticsearchException.guessRootCauses(ElasticsearchException.java:699)	
    	at org.elasticsearch.server@8.12.2/org.elasticsearch.ElasticsearchException.generateFailureXContent(ElasticsearchException.java:627)	
    	at org.elasticsearch.server@8.12.2/org.elasticsearch.rest.RestResponse.build(RestResponse.java:180)	
    	at org.elasticsearch.server@8.12.2/org.elasticsearch.rest.RestResponse.<init>(RestResponse.java:140)	
    	at org.elasticsearch.server@8.12.2/org.elasticsearch.rest.RestResponse.<init>(RestResponse.java:111)	
    	at org.elasticsearch.server@8.12.2/org.elasticsearch.rest.action.RestActionListener.onFailure(RestActionListener.java:55)	
    	at org.elasticsearch.xpack.esql.action.EsqlResponseListener.lambda$wrapWithLogging$1(EsqlResponseListener.java:168)	
    	at org.elasticsearch.server@8.12.2/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:62)	
    	at org.elasticsearch.server@8.12.2/org.elasticsearch.action.ActionListener$2.onFailure(ActionListener.java:186)	
    	at org.elasticsearch.server@8.12.2/org.elasticsearch.rest.action.RestCancellableNodeClient$1.onFailure(RestCancellableNodeClient.java:96)	
    	at org.elasticsearch.server@8.12.2/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:62)	
    	at org.elasticsearch.server@8.12.2/org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:73)	
    	at org.elasticsearch.server@8.12.2/org.elasticsearch.action.ActionListener$4.onFailure(ActionListener.java:354)	
    	at org.elasticsearch.server@8.12.2/org.elasticsearch.tasks.TaskManager$1.onFailure(TaskManager.java:214)	
    	at org.elasticsearch.server@8.12.2/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:62)	
    	at org.elasticsearch.server@8.12.2/org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:73)	
    	at org.elasticsearch.server@8.12.2/org.elasticsearch.action.DelegatingActionListener.onFailure(DelegatingActionListener.java:27)	
    	at org.elasticsearch.xpack.esql.plugin.ComputeService.lambda$execute$1(ComputeService.java:144)	
    	at org.elasticsearch.server@8.12.2/org.elasticsearch.action.ActionListenerImplementations$DelegatingResponseActionListener.acceptException(ActionListenerImplementations.java:186)	
    	at org.elasticsearch.server@8.12.2/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:62)	
    	at org.elasticsearch.server@8.12.2/org.elasticsearch.action.ActionListenerImplementations$DelegatingResponseActionListener.onFailure(ActionListenerImplementations.java:191)	
    	at org.elasticsearch.server@8.12.2/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:62)	
    	at org.elasticsearch.server@8.12.2/org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:73)	
    	at org.elasticsearch.server@8.12.2/org.elasticsearch.action.DelegatingActionListener.onFailure(DelegatingActionListener.java:27)	
    	at org.elasticsearch.server@8.12.2/org.elasticsearch.action.ActionListenerImplementations$RunBeforeActionListener.onFailure(ActionListenerImplementations.java:308)	
    	at org.elasticsearch.server@8.12.2/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:62)	
    	at org.elasticsearch.server@8.12.2/org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:73)	
    	at org.elasticsearch.server@8.12.2/org.elasticsearch.action.ActionListener$4.onFailure(ActionListener.java:354)	
    	at org.elasticsearch.server@8.12.2/org.elasticsearch.action.ActionLyistenerImplementations.safeAcceptException(ActionListenerImplementations.java:62)	
    	at org.elasticsearch.server@8.12.2/org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:73)	
    	at org.elasticsearch.server@8.12.2/org.elasticsearch.action.DelegatingActionListener.onFailure(DelegatingActionListener.java:27)	
    	at org.elasticsearch.server@8.12.2/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:62)	
    	at org.elasticsearch.server@8.12.2/org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:73)	
    	at org.elasticsearch.server@8.12.2/org.elasticsearch.action.ActionListener$4.onFailure(ActionListener.java:354)	
    	at org.elasticsearch.server@8.12.2/org.elasticsearch.action.support.RefCountingListener.finish(RefCountingListener.java:127)	
    	at org.elasticsearch.base@8.12.2/org.elasticsearch.core.AbstractRefCounted$1.closeInternal(AbstractRefCounted.java:118)	
    	at org.elasticsearch.base@8.12.2/org.elasticsearch.core.AbstractRefCounted.decRef(AbstractRefCounted.java:70)	
    	at org.elasticsearch.server@8.12.2/org.elasticsearch.action.support.RefCountingRunnable.close(RefCountingRunnable.java:112)	
    	at org.elasticsearch.base@8.12.2/org.elasticsearch.core.Releasables$3.close(Releasables.java:142)	
    	at org.elasticsearch.server@8.12.2/org.elasticsearch.action.support.RefCountingListener$1.onFailure(RefCountingListener.java:158)	
    	at org.elasticsearch.server@8.12.2/org.elasticsearch.action.support.SubscribableListener$FailureResult.complete(SubscribableListener.java:322)	
    	at org.elasticsearch.server@8.12.2/org.elasticsearch.action.support.SubscribableListener.tryComplete(SubscribableListener.java:234)	
    	at org.elasticsearch.server@8.12.2/org.elasticsearch.action.support.SubscribableListener.setResult(SubscribableListener.java:275)	
    	at org.elasticsearch.server@8.12.2/org.elasticsearch.action.support.SubscribableListener.onFailure(SubscribableListener.java:178)	
    	at org.elasticsearch.compute.operator.exchange.ExchangeSourceHandler.closeInternal(ExchangeSourceHandler.java:244)	
    	at org.elasticsearch.base@8.12.2/org.elasticsearch.core.AbstractRefCounted.decRef(AbstractRefCounted.java:70)	
    	at org.elasticsearch.xpack.esql.plugin.ComputeService.lambda$execute$7(ComputeService.java:234)	
    	at org.elasticsearch.server@8.12.2/org.elasticsearch.action.ActionListenerImplementations$ResponseWrappingActionListener.onResponse(ActionListenerImplementations.java:236)	
    	at org.elasticsearch.server@8.12.2/org.elasticsearch.action.ActionListenerImplementations$MappedActionListener.onResponse(ActionListenerImplementations.java:95)	
    	at org.elasticsearch.server@8.12.2/org.elasticsearch.action.support.ContextPreservingActionListener.onResponse(ContextPreservingActionListener.java:32)	
    	at org.elasticsearch.server@8.12.2/org.elasticsearch.action.ActionListenerResponseHandler.handleResponse(ActionListenerResponseHandler.java:49)	
    	at org.elasticsearch.server@8.12.2/org.elasticsearch.transport.TransportService$UnregisterChildTransportResponseHandler.handleResponse(TransportService.java:1709)	
    	at org.elasticsearch.server@8.12.2/org.elasticsearch.transport.TransportService$ContextRestoreResponseHandler.handleResponse(TransportService.java:1425)	
    	at org.elasticsearch.server@8.12.2/org.elasticsearch.transport.TransportService$DirectResponseChannel.processResponse(TransportService.java:1524)	
    	at org.elasticsearch.server@8.12.2/org.elasticsearch.transport.TransportService$DirectResponseChannel$1.doRun(TransportService.java:1501)	
    	at org.elasticsearch.server@8.12.2/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)	
    	at org.elasticsearch.server@8.12.2/org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:33)	
    	at org.elasticsearch.server@8.12.2/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:983)	
    	at org.elasticsearchy.server@8.12.2/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)	
    	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)	
    	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)	
    	at java.base/java.lang.Thread.run(Thread.java:1583)	
    Caused by: java.lang.NullPointerException: Cannot invoke "org.elasticsearch.xpack.esql.planner.Layout$ChannelSet.nameIds()" because the return value of "java.util.List.get(int)" is null	
    	at org.elasticsearch.xpack.esql.planner.ExchangeLayout.get(ExchangeLayout.java:36)	
    	at org.elasticsearch.xpack.esql.planner.AbstractPhysicalOperationProviders.groupingPhysicalOperation(AbstractPhysicalOperationProviders.java:124)	
    	at org.elasticsearch.xpack.esql.planner.LocalExecutionPlanner.planAggregation(LocalExecutionPlanner.java:237)	
    	at org.elasticsearch.xpack.esql.planner.LocalExecutionPlanner.plan(LocalExecutionPlanner.java:185)	
    	at org.elasticsearch.xpack.esql.planner.LocalExecutionPlanner.planProject(LocalExecutionPlanner.java:520)	
    	at org.elasticsearch.xpack.esql.planner.LocalExecutionPlanner.plan(LocalExecutionPlanner.java:199)	
    	at org.elasticsearch.xpack.esql.planner.LocalExecutionPlanner.planTopN(LocalExecutionPlanner.java:347)	
    	at org.elasticsearch.xpack.esql.planner.LocalExecutionPlanner.plan(LocalExecutionPlanner.java:191)	
    	at org.elasticsearch.xpack.esql.planner.LocalExecutionPlanner.planOutput(LocalExecutionPlanner.java:280)	
    	at org.elasticsearch.xpack.esql.planner.LocalExecutionPlanner.plan(LocalExecutionPlanner.java:227)	
    	at org.elasticsearch.xpack.esql.planner.LocalExecutionPlanner.plan(LocalExecutionPlanner.java:170)	
    	at org.elasticsearch.xpack.esql.plugin.ComputeService.runCompute(ComputeService.java:312)	
    	at org.elasticsearch.xpack.esql.plugin.ComputeService.lambda$execute$7(ComputeService.java:206)	
    	... 15 more	
    	Suppressed: org.elasticsearch.transport.SendRequestTransportException: [test-cluster-0][127.0.0.1:44201][indices:data/read/esql/data]	
    		at org.elasticsearch.server@8.12.2/org.elasticsearch.transport.TransportService.sendRequest(TransportService.java:851)	
    		at org.elasticsearch.server@8.12.2/org.elasticsearch.transport.TransportService.sendChildRequest(TransportService.java:911)	
    		at org.elasticsearch.server@8.12.2/org.elasticsearch.transport.TransportService.sendChildRequest(TransportService.java:888)	
    		at org.elasticsearch.xpack.esql.plugin.ComputeService.lambda$runComputeOnRemoteNodes$11(ComputeService.java:269)	
    		at org.elasticsearch.server@8.12.2/org.elasticsearch.action.ActionListenerImplementations$ResponseWrappingActionListener.onResponse(ActionListenerImplementations.java:236)	
    		at org.elasticsearch.server@8.12.2/org.elasticsearch.action.ActionListenerImplementations$MappedActionListener.onResponse(ActionListenerImplementations.java:95)	
    		at org.elasticsearch.server@8.12.2/org.elasticsearch.action.ActionListenerResponseHandler.handleResponse(ActionListenerResponseHandler.java:49)	
    		... 10 more	
    	Caused by: org.elasticsearch.tasks.TaskCancelledException: parent task was cancelled [cancelled]	
    		at org.elasticsearch.server@8.12.2/org.elasticsearch.tasks.TaskManager$CancellableTaskHolder.registerChildConnection(TaskManager.java:666)	
    		at org.elasticsearch.server@8.12.2/org.elasticsearch.tasks.TaskManager.registerChildConnection(TaskManager.java:362)	
    		at org.elasticsearch.server@8.12.2/org.elasticsearch.transport.TransportService.sendRequest(TransportService.java:831)	
    		... 16 more	
    	Suppressed: org.elasticsearch.transport.SendRequestTransportException: [test-cluster-0][127.0.0.1:44201][internal:data/read/esql/exchange]	
    		at org.elasticsearch.server@8.12.2/org.elasticsearch.transport.TransportService.sendRequest(TransportService.java:851)	
    		at org.elasticsearch.server@8.12.2/org.elasticsearch.transport.TransportService.sendChildRequest(TransportService.java:911)	
    		at org.elasticsearch.server@8.12.2/org.elasticsearch.transport.TransportService.sendChildRequest(Transpoy~rtService.java:888)	
    		at org.elasticsearch.compute.operator.exchange.ExchangeService$TransportRemoteSink.fetchPageAsync(ExchangeService.java:271)	
    		at org.elasticsearch.compute.operator.exchange.ExchangeSourceHandler$RemoteSinkFetcher.fetchPage(ExchangeSourceHandler.java:159)	
    		at org.elasticsearch.compute.operator.exchange.ExchangeSourceHandler$1.doRun(ExchangeSourceHandler.java:234)	
    		... 7 more	
    	Caused by: org.elasticsearch.tasks.TaskCancelledException: parent task was cancelled [cancelled]	
    		at org.elasticsearch.server@8.12.2/org.elasticsearch.tasks.TaskManager$CancellableTaskHolder.registerChildConnection(TaskManager.java:666)	
    		at org.elasticsearch.server@8.12.2/org.elasticsearch.tasks.TaskManager.registerChildConnection(TaskManager.java:362)	
    		at org.elasticsearch.server@8.12.2/org.elasticsearch.transport.TransportService.sendRequest(TransportService.java:831)	
    		... 12 more	
    ÿÿÿdtypevnull_pointer_exceptionfreasonxCannot invoke "org.elasticsearch.xpack.esql.planner.Layout$ChannelSet.nameIds()" because the return value of "java.util.List.get(int)" is nullkstack_traceyjava.lang.NullPointerException: Cannot invoke "org.elasticsearch.xpack.esql.planner.Layout$ChannelSet.nameIds()" because the return value of "java.util.List.get(int)" is null	
    	at org.elasticsearch.xpack.esql.planner.ExchangeLayout.get(ExchangeLayout.java:36)	
    	at org.elasticsearch.xpack.esql.planner.AbstractPhysicalOperationProviders.groupingPhysicalOperation(AbstractPhysicalOperationProviders.java:124)	
    	at org.elasticsearch.xpack.esql.planner.LocalExecutionPlanner.planAggregation(LocalExecutionPlanner.java:237)	
    	at org.elasticsearch.xpack.esql.planner.LocalExecutionPlanner.plan(LocalExecutionPlanner.java:185)	
    	at org.elasticsearch.xpack.esql.planner.LocalExecutionPlanner.planProject(LocalExecutionPlanner.java:520)	
    	at org.elasticsearch.xpack.esql.planner.LocalExecutionPlanner.plan(LocalExecutionPlanner.java:199)	
    	at org.elasticsearch.xpack.esql.planner.LocalExecutionPlanner.planTopN(LocalExecutionPlanner.java:347)	
    	at org.elasticsearch.xpack.esql.planner.LocalExecutionPlanner.plan(LocalExecutionPlanner.java:191)	
    	at org.elasticsearch.xpack.esql.planner.LocalExecutionPlanner.planOutput(LocalExecutionPlanner.java:280)	
    	at org.elasticsearch.xpack.esql.planner.LocalExecutionPlanner.plan(LocalExecutionPlanner.java:227)	
    	at org.elasticsearch.xpack.esql.planner.LocalExecutionPlanner.plan(LocalExecutionPlanner.java:170)	
    	at org.elasticsearch.xpack.esql.plugin.ComputeService.runCompute(ComputeService.java:312)	
    	at org.elasticsearch.xpack.esql.plugin.ComputeService.lambda$execute$7(ComputeService.java:206)	
    	at org.elasticsearch.server@8.12.2/org.elasticsearch.action.ActionListenerImplementations$ResponseWrappingActionListener.onResponse(ActionListenerImplementations.java:236)	
    	at org.elasticsearch.server@8.12.2/org.elasticsearch.action.ActionListenerImplementations$MappedActionListener.onResponse(ActionListenerImplementations.java:95)	
    	at org.elasticsearch.server@8.12.2/org.elasticsearch.action.support.ContextPreservingActionListener.onResponse(ContextPreservingActionListener.java:32)	
    	at org.elasticsearch.server@8.12.2/org.elasticsearch.action.ActionListenerResponseHandler.handleResponse(ActionListenerResponseHandler.java:49)	
    	at org.elasticsearch.server@8.12.2/org.elasticsearch.transport.TransportService$UnregisterChildTransportResponseHandler.handleResponse(TransportService.java:1709)	
    	at org.elasticsearch.server@8.12.2/org.elasticsearch.transport.TransportService$ContextRestoreResponseHandler.handleResponse(TransportService.java:1425)	
    	at org.elasticsearch.server@8.12.2/org.elasticsearch.transport.TransportService$DirectResponseChannel.processResponse(TransportService.java:1524)	
    	at org.elasticsearch.server@8.12.2/org.elasticsearch.transport.TransportService$DirectResponseChannel$1.doRun(TransportService.java:1501)	
    	at org.elasticsearch.server@8.12.2/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)	
    	at org.elasticsearch.server@8.12.2/org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:33)	
    	at org.elasticsearch.server@8.12.2/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:983)	
    	at org.elasticsearch.server@8.12.2/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)	
    	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)	
    	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)	
    	at java.base/java.lang.Thread.run(Thread.java:1583)	
    	Suppressed: org.elasticsearch.transport.SendRequestTransportException: [test-cluster-0][127.0.0.1:44201][indices:data/read/esql/data]	
    		at org.elasticsearch.server@8.12.2/org.elasticsearch.transport.TransportService.sendRequest(TransportService.java:851)	
    		at org.elasticsearch.server@8.12.2/org.elasticsearch.transport.TransportService.sendChildRequest(TransportService.java:911)	
    		at org.elasticsearch.server@8.12.2/org.elasticsearch.transport.TransportService.sendChildy	ßRequest(TransportService.java:888)	
    		at org.elasticsearch.xpack.esql.plugin.ComputeService.lambda$runComputeOnRemoteNodes$11(ComputeService.java:269)	
    		at org.elasticsearch.server@8.12.2/org.elasticsearch.action.ActionListenerImplementations$ResponseWrappingActionListener.onResponse(ActionListenerImplementations.java:236)	
    		at org.elasticsearch.server@8.12.2/org.elasticsearch.action.ActionListenerImplementations$MappedActionListener.onResponse(ActionListenerImplementations.java:95)	
    		at org.elasticsearch.server@8.12.2/org.elasticsearch.action.ActionListenerResponseHandler.handleResponse(ActionListenerResponseHandler.java:49)	
    		... 10 more	
    	Caused by: org.elasticsearch.tasks.TaskCancelledException: parent task was cancelled [cancelled]	
    		at org.elasticsearch.server@8.12.2/org.elasticsearch.tasks.TaskManager$CancellableTaskHolder.registerChildConnection(TaskManager.java:666)	
    		at org.elasticsearch.server@8.12.2/org.elasticsearch.tasks.TaskManager.registerChildConnection(TaskManager.java:362)	
    		at org.elasticsearch.server@8.12.2/org.elasticsearch.transport.TransportService.sendRequest(TransportService.java:831)	
    		... 16 more	
    	Suppressed: org.elasticsearch.transport.SendRequestTransportException: [test-cluster-0][127.0.0.1:44201][internal:data/read/esql/exchange]	
    		at org.elasticsearch.server@8.12.2/org.elasticsearch.transport.TransportService.sendRequest(TransportService.java:851)	
    		at org.elasticsearch.server@8.12.2/org.elasticsearch.transport.TransportService.sendChildRequest(TransportService.java:911)	
    		at org.elasticsearch.server@8.12.2/org.elasticsearch.transport.TransportService.sendChildRequest(TransportService.java:888)	
    		at org.elasticsearch.compute.operator.exchange.ExchangeService$TransportRemoteSink.fetchPageAsync(ExchangeService.java:271)	
    		at org.elasticsearch.compute.operator.exchange.ExchangeSourceHandler$RemoteSinkFetcher.fetchPage(ExchangeSourceHandler.java:159)	
    		at org.elasticsearch.compute.operator.exchange.ExchangeSourceHandler$1.doRun(ExchangeSourceHandler.java:234)	
    		... 7 more	
    	Caused by: org.elasticsearch.tasks.TaskCancelledException: parent task was cancelled [cancelled]	
    		at org.elasticsearch.server@8.12.2/org.elasticsearch.tasks.TaskManager$CancellableTaskHolder.registerChildConnection(TaskManager.java:666)	
    		at org.elasticsearch.server@8.12.2/org.elasticsearch.tasks.TaskManager.registerChildConnection(TaskManager.java:362)	
    		at org.elasticsearch.server@8.12.2/org.elasticsearch.transport.TransportService.sendRequest(TransportService.java:831)	
    		... 12 more
```
Here's the failure history for this test suite: https://es-delivery-stats.elastic.dev/app/r/s/DMnCM

**Build scan:**
https://gradle-enterprise.elastic.co/s/gohlvellh4eae/tests/:x-pack:plugin:esql:qa:server:mixed-cluster:v8.12.2%23javaRestTest/org.elasticsearch.xpack.esql.qa.mixed.FieldExtractorIT/testPropagateUnsupportedToSubFields


**Reproduction line:**
```
./gradlew ':x-pack:plugin:esql:qa:server:mixed-cluster:v8.12.2#javaRestTest' -Dtests.class="org.elasticsearch.xpack.esql.qa.mixed.FieldExtractorIT" -Dtests.method="testPropagateUnsupportedToSubFields" -Dtests.seed=DFB20F126AD53B9F -Dtests.bwc=true -Dtests.locale=nb-NO -Dtests.timezone=Antarctica/Syowa -Druntime.java=21
```

**Applicable branches:**
main

**Reproduces locally?:**
Didn't try

**Failure history:**
[Failure dashboard for `org.elasticsearch.xpack.esql.qa.mixed.FieldExtractorIT#testPropagateUnsupportedToSubFields`](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('testPropagateUnsupportedToSubFields'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.xpack.esql.qa.mixed.FieldExtractorIT'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))


**Failure excerpt:**
```
java.lang.AssertionError: 
Expected: a string containing "Cannot use field [f] with unsupported type [ip_range]"
     but: was ":)
úerrorúroot_causeøútypeXx_content_parse_exceptionreasonk[-1:48] [esql/query] unknown field [version]stack_traceäorg.elasticsearch.ElasticsearchException$1: [-1:48] [esql/query] unknown field [version]
	at org.elasticsearch.server@8.12.2/org.elasticsearch.ElasticsearchException.guessRootCauses(ElasticsearchException.java:699)
	at org.elasticsearch.server@8.12.2/org.elasticsearch.ElasticsearchException.generateFailureXContent(ElasticsearchException.java:627)
	at org.elasticsearch.server@8.12.2/org.elasticsearch.rest.RestResponse.build(RestResponse.java:180)
	at org.elasticsearch.server@8.12.2/org.elasticsearch.rest.RestResponse.<init>(RestResponse.java:140)
	at org.elasticsearch.server@8.12.2/org.elasticsearch.rest.RestResponse.<init>(RestResponse.java:111)
	at org.elasticsearch.server@8.12.2/org.elasticsearch.rest.RestController.dispatchRequest(RestController.java:443)
	at org.elasticsearch.server@8.12.2/org.elasticsearch.rest.RestController.tryAllHandlers(RestController.java:570)
	at org.elasticsearch.server@8.12.2/org.elasticsearch.rest.RestController.dispatchRequest(RestController.java:325)
	at org.elasticsearch.server@8.12.2/org.elasticsearch.http.AbstractHttpServerTransport.dispatchRequest(AbstractHttpServerTransport.java:458)
	at org.elasticsearch.server@8.12.2/org.elasticsearch.http.AbstractHttpServerTransport.handleIncomingRequest(AbstractHttpServerTransport.java:554)
	at org.elasticsearch.server@8.12.2/org.elasticsearch.http.AbstractHttpServerTransport.incomingRequest(AbstractHttpServerTransport.java:431)
	at org.elasticsearch.transport.netty4@8.12.2/org.elasticsearch.http.netty4.Netty4HttpPipeliningHandler.handlePipelinedRequest(Netty4HttpPipeliningHandler.java:128)
	at org.elasticsearch.transport.netty4@8.12.2/org.elasticsearch.http.netty4.Netty4HttpPipeliningHandler.channelRead(Netty4HttpPipeliningHandler.java:118)
	at io.netty.transport@4.1.94.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442)
	at io.netty.transport@4.1.94.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.transport@4.1.94.Final/io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
	at io.netty.codec@4.1.94.Final/io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103)
	at io.netty.codec@4.1.94.Final/io.netty.handler.codec.MessageToMessageCodec.channelRead(MessageToMessageCodec.java:111)
	at io.netty.transport@4.1.94.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442)
	at io.netty.transport@4.1.94.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.transport@4.1.94.Final/io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
	at io.netty.codec@4.1.94.Final/io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103)
	at io.netty.transport@4.1.94.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
	at io.netty.transport@4.1.94.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.transport@4.1.94.Final/io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
	at io.netty.codec@4.1.94.Final/io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103)
	at io.netty.transport@4.1.94.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
	at io.netty.transport@4.1.94.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.transport@4.1.94.Final/io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
	at io.netty.codec@4.1.94.Final/io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:346)
	at io.netty.codec@4.1.94.Final/io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:318)
	at io.netty.transport@4.1.94.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
	at io.netty.transport@4.1.94.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.transport@4.1.94.Final/io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
	at io.netty.codec@4.1.94.Final/io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103)
	at io.netty.transport@4.1.94.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
	at io.netty.transport@4.1.94.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.transport@4.1.94.Final/io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
	at io.netty.transport@4.1.94.Final/io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410)
	at io.netty.transport@4.1.94.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:440)
	at io.netty.transport@4.1.94.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.transport@4.1.94.Final/io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919)
	at io.netty.transport@4.1.94.Final/io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:166)
	at io.netty.transport@4.1.94.Final/io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:788)
	at io.netty.transport@4.1.94.Final/io.netty.channel.nio.NioEventLoop.processSelectedKeysPlain(NioEventLoop.java:689)
	at io.netty.transport@4.1.94.Final/io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:652)
	at io.netty.transport@4.1.94.Final/io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562)
	at io.netty.common@4.1.94.Final/io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
	at io.netty.common@4.1.94.Final/io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: org.elasticsearch.xcontent.XContentParseException: [-1:48] [esql/query] unknown field [version]
	at org.elasticsearch.xcontent@8.12.2/org.elasticsearch.xcontent.ObjectParser.lambda$errorOnUnknown$2(ObjectParser.java:105)
	at org.elasticsearch.xcontent@8.12.2/org.elasticsearch.xcontent.ObjectParser.parse(ObjectParser.java:299)
	at org.elasticsearch.xcontent@8.12.2/org.elasticsearch.xcontent.ObjectParser.parse(ObjectParser.java:259)
	at org.elasticsearch.xcontent@8.12.2/org.elasticsearch.xcontent.ObjectParser.apply(ObjectParser.java:382)
	at org.elasticsearch.xpack.esql.action.EsqlQueryRequest.fromXContent(EsqlQueryRequest.java:159)
	at org.elasticsearch.xpack.esql.action.RestEsqlQueryAction.prepareRequest(RestEsqlQueryAction.java:51)
	at org.elasticsearch.server@8.12.2/org.elasticsearch.rest.BaseRestHandler.handleRequest(BaseRestHandler.java:80)
	at org.elasticsearch.security@8.12.2/org.elasticsearch.xpack.security.rest.SecurityRestFilter.doHandleRequest(SecurityRestFilter.java:94)
	at org.elasticsearch.security@8.12.2/org.elasticsearch.xpack.security.rest.SecurityRestFilter.handleRequest(SecurityRestFilter.java:73)
	at org.elasticsearch.server@8.12.2/org.elasticsearch.rest.RestController.dispatchRequest(RestController.java:441)
	... 44 more
üûùBXx_content_parse_exceptionCk[-1:48] [esql/query] unknown field [version]Däorg.elasticsearch.xcontent.XContentParseException: [-1:48] [esql/query] unknown field [version]
	at org.elasticsearch.xcontent@8.12.2/org.elasticsearch.xcontent.ObjectParser.lambda$errorOnUnknown$2(ObjectParser.java:105)
	at org.elasticsearch.xcontent@8.12.2/org.elasticsearch.xcontent.ObjectParser.parse(ObjectParser.java:299)
	at org.elasticsearch.xcontent@8.12.2/org.elasticsearch.xcontent.ObjectParser.parse(ObjectParser.java:259)
	at org.elasticsearch.xcontent@8.12.2/org.elasticsearch.xcontent.ObjectParser.apply(ObjectParser.java:382)
	at org.elasticsearch.xpack.esql.action.EsqlQueryRequest.fromXContent(EsqlQueryRequest.java:159)
	at org.elasticsearch.xpack.esql.action.RestEsqlQueryAction.prepareRequest(RestEsqlQueryAction.java:51)
	at org.elasticsearch.server@8.12.2/org.elasticsearch.rest.BaseRestHandler.handleRequest(BaseRestHandler.java:80)
	at org.elasticsearch.security@8.12.2/org.elasticsearch.xpack.security.rest.SecurityRestFilter.doHandleRequest(SecurityRestFilter.java:94)
	at org.elasticsearch.security@8.12.2/org.elasticsearch.xpack.security.rest.SecurityRestFilter.handleRequest(SecurityRestFilter.java:73)
	at org.elasticsearch.server@8.12.2/org.elasticsearch.rest.RestController.dispatchRequest(RestController.java:441)
	at org.elasticsearch.server@8.12.2/org.elasticsearch.rest.RestController.tryAllHandlers(RestController.java:570)
	at org.elasticsearch.server@8.12.2/org.elasticsearch.rest.RestController.dispatchRequest(RestController.java:325)
	at org.elasticsearch.server@8.12.2/org.elasticsearch.http.AbstractHttpServerTransport.dispatchRequest(AbstractHttpServerTransport.java:458)
	at org.elasticsearch.server@8.12.2/org.elasticsearch.http.AbstractHttpServerTransport.handleIncomingRequest(AbstractHttpServerTransport.java:554)
	at org.elasticsearch.server@8.12.2/org.elasticsearch.http.AbstractHttpServerTransport.incomingRequest(AbstractHttpServerTransport.java:431)
	at org.elasticsearch.transport.netty4@8.12.2/org.elasticsearch.http.netty4.Netty4HttpPipeliningHandler.handlePipelinedRequest(Netty4HttpPipeliningHandler.java:128)
	at org.elasticsearch.transport.netty4@8.12.2/org.elasticsearch.http.netty4.Netty4HttpPipeliningHandler.channelRead(Netty4HttpPipeliningHandler.java:118)
	at io.netty.transport@4.1.94.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442)
	at io.netty.transport@4.1.94.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.transport@4.1.94.Final/io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
	at io.netty.codec@4.1.94.Final/io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103)
	at io.netty.codec@4.1.94.Final/io.netty.handler.codec.MessageToMessageCodec.channelRead(MessageToMessageCodec.java:111)
	at io.netty.transport@4.1.94.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442)
	at io.netty.transport@4.1.94.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.transport@4.1.94.Final/io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
	at io.netty.codec@4.1.94.Final/io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103)
	at io.netty.transport@4.1.94.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
	at io.netty.transport@4.1.94.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.transport@4.1.94.Final/io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
	at io.netty.codec@4.1.94.Final/io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103)
	at io.netty.transport@4.1.94.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
	at io.netty.transport@4.1.94.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.transport@4.1.94.Final/io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
	at io.netty.codec@4.1.94.Final/io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:346)
	at io.netty.codec@4.1.94.Final/io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:318)
	at io.netty.transport@4.1.94.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
	at io.netty.transport@4.1.94.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.transport@4.1.94.Final/io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
	at io.netty.codec@4.1.94.Final/io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103)
	at io.netty.transport@4.1.94.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
	at io.netty.transport@4.1.94.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.transport@4.1.94.Final/io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
	at io.netty.transport@4.1.94.Final/io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410)
	at io.netty.transport@4.1.94.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:440)
	at io.netty.transport@4.1.94.Final/io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.transport@4.1.94.Final/io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919)
	at io.netty.transport@4.1.94.Final/io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:166)
	at io.netty.transport@4.1.94.Final/io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:788)
	at io.netty.transport@4.1.94.Final/io.netty.channel.nio.NioEventLoop.processSelectedKeysPlain(NioEventLoop.java:689)
	at io.netty.transport@4.1.94.Final/io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:652)
	at io.netty.transport@4.1.94.Final/io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562)
	at io.netty.common@4.1.94.Final/io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
	at io.netty.common@4.1.94.Final/io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at java.base/java.lang.Thread.run(Thread.java:1583)
üûstatus$ û"

  at __randomizedtesting.SeedInfo.seed([DFB20F126AD53B9F:BE7EEA2188CC12E8]:0)
  at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
  at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:6)
  at org.elasticsearch.test.ESTestCase.assertThat(ESTestCase.java:2151)
  at org.elasticsearch.xpack.esql.qa.rest.FieldExtractorTestCase.testPropagateUnsupportedToSubFields(FieldExtractorTestCase.java:819)
  at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
  at java.lang.reflect.Method.invoke(Method.java:580)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.junit.rules.RunRules.evaluate(RunRules.java:20)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:48)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.junit.rules.RunRules.evaluate(RunRules.java:20)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.elasticsearch.test.cluster.local.DefaultLocalElasticsearchCluster$1.evaluate(DefaultLocalElasticsearchCluster.java:47)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at org.junit.rules.RunRules.evaluate(RunRules.java:20)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
  at java.lang.Thread.run(Thread.java:1583)

```

--- a/x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/EsqlSpecTestCase.java
+++ b/x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/EsqlSpecTestCase.java
@@ -71,7 +71,6 @@ public abstract class EsqlSpecTestCase extends ESRestTestCase {
 
     public static Set<EsqlVersion> availableVersions() {
         if ("true".equals(System.getProperty("tests.version_parameter_unsupported"))) {
-            // TODO: skip tests with explicitly set version and/or strip the version if it's 2024.04.01.
             return Set.of();
         }
         return Build.current().isSnapshot() ? Set.of(EsqlVersion.values()) : Set.of(EsqlVersion.releasedAscending());
@@ -157,6 +156,7 @@ public abstract class EsqlSpecTestCase extends ESRestTestCase {
         RequestObjectBuilder builder = new RequestObjectBuilder(randomFrom(XContentType.values()));
 
         String versionString = null;
+        // TODO: skip tests with explicitly set version and/or strip the version if it's 2024.04.01.
         if (availableVersions().isEmpty() == false) {
             EsqlVersion version = randomFrom(availableVersions());
             versionString = randomBoolean() ? version.toString() : version.versionStringWithoutEmoji();

--- a/x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/FieldExtractorTestCase.java
+++ b/x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/FieldExtractorTestCase.java
@@ -26,7 +26,7 @@ import org.elasticsearch.test.rest.ESRestTestCase;
 import org.elasticsearch.xcontent.XContentBuilder;
 import org.elasticsearch.xcontent.XContentType;
 import org.elasticsearch.xcontent.json.JsonXContent;
-import org.elasticsearch.xpack.esql.EsqlTestUtils;
+import org.elasticsearch.xpack.esql.version.EsqlVersion;
 import org.hamcrest.Matcher;
 import org.junit.Before;
 
@@ -39,6 +39,7 @@ import java.util.Comparator;
 import java.util.List;
 import java.util.Locale;
 import java.util.Map;
+import java.util.Set;
 import java.util.TreeMap;
 import java.util.function.Function;
 
@@ -1435,7 +1436,14 @@ public abstract class FieldExtractorTestCase extends ESRestTestCase {
     }
 
     private static Map<String, Object> runEsql(String query) throws IOException {
-        return runEsqlSync(new RestEsqlTestCase.RequestObjectBuilder().query(query).version(EsqlTestUtils.latestEsqlVersionOrSnapshot()));
+        String versionString = null;
+        Set<EsqlVersion> versions = EsqlSpecTestCase.availableVersions();
+        if (versions.isEmpty() == false) {
+            // Use the latest released version or SNAPSHOT, if available.
+            versionString = versions.stream().max(Comparator.comparingInt(EsqlVersion::id)).toString();
+        }
+
+        return runEsqlSync(new RestEsqlTestCase.RequestObjectBuilder().query(query).version(versionString));
     }
 
 }

--- a/x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/FieldExtractorTestCase.java
+++ b/x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/FieldExtractorTestCase.java
@@ -39,7 +39,6 @@ import java.util.Comparator;
 import java.util.List;
 import java.util.Locale;
 import java.util.Map;
-import java.util.Set;
 import java.util.TreeMap;
 import java.util.function.Function;
 
@@ -1436,12 +1435,12 @@ public abstract class FieldExtractorTestCase extends ESRestTestCase {
     }
 
     private static Map<String, Object> runEsql(String query) throws IOException {
-        String versionString = null;
-        Set<EsqlVersion> versions = EsqlSpecTestCase.availableVersions();
-        if (versions.isEmpty() == false) {
-            // Use the latest released version or SNAPSHOT, if available.
-            versionString = versions.stream().max(Comparator.comparingInt(EsqlVersion::id)).toString();
-        }
+        // Use the latest released version or SNAPSHOT, if available.
+        String versionString = EsqlSpecTestCase.availableVersions()
+            .stream()
+            .max(Comparator.comparingInt(EsqlVersion::id))
+            .map(EsqlVersion::toString)
+            .orElse(null);
 
         return runEsqlSync(new RestEsqlTestCase.RequestObjectBuilder().query(query).version(versionString));
     }

--- a/x-pack/plugin/esql/qa/testFixtures/src/main/resources/stats.csv-spec
+++ b/x-pack/plugin/esql/qa/testFixtures/src/main/resources/stats.csv-spec
@@ -1552,7 +1552,7 @@ s2point1:d | s_mv:i | languages:i
 2.1        | 3      | null
 ;
 
-evalOverridingKey
+evalOverridingKey#[skip:-8.13.1,reason:fixed in 8.13.2]
 FROM employees
 | EVAL k = languages
 | STATS c = COUNT() BY languages, k

